test/node-rpc-test.js

```diff
@@ -68,11 +70,93 @@ describe('RPC', function() {
     assert.deepEqual(info.localservicenames, ['NETWORK', 'WITNESS']);
   });
 
+  it('should rpc getblockhash', async () => {
+    const info = await nclient.execute('getblockhash', [node.chain.tip.height]);
+    assert.strictEqual(util.revHex(node.chain.tip.hash), info);
+  });
+
+  describe('Blockchain', function () {
+    it('should rpc getchaintips', async () => {
+      const info = await nclient.execute('getchaintips', []);
+      assert.strictEqual(info.length, 1);
+      assert.strictEqual(util.revHex(node.chain.tip.hash), info[0].hash);
+    });
+
+    it('should rpc getchaintips for chain fork', async () => {
+      // Get chain entry associated with genesis block (height == 0)
+      let entry1 = await node.chain.getEntry(0);
+     // extnding chain1 from genesis.
+      for (let i = 0; i <= 3; i++) {
+        const block = await node.miner.mineBlock(entry1);
+        entry1 = await node.chain.add(block);
+      }
+      // current state:
+      //        genesis block -- block01 -- block02 -- block03
+      // Creting a chain fork, by mining block again on genesis as parent.
+      let entry2 = await node.chain.getEntry(0);
+      for (let i = 0; i <= 2; i++) {
+        const block = await node.miner.mineBlock(entry2);
+        entry2 = await node.chain.add(block);
+      }
+      // current state:
+      //                        block01 -- block02 -- block03 (chain1, with height 3)
+      //                      /
+      //         genesis block
+      //                      \
+      //                        block01 -- block02 (chain2, with height 2)
+      const info = await nclient.execute('getchaintips', []);
+      assert.notEqual(entry1.hash, entry2.hash);
+      assert.strictEqual(info.length, 2);
+      assert.strictEqual(node.chain.tip.hash, entry1.hash);
```

Comment:
Here is what I was thinking for this. We want to check the entire output of the RPC command, we want to make sure both of the chain tips we know about are returned but they could be returned in either order so we have to be a little ugly about it. I also suggested an update to the ASCII art here just to keep lines below 80 columns and also fix the "height" values.

```suggestion
      // current state:
      //                  block01 -- block02 -- block03 (entry1, with height 4)
      //                /
      //   genesis block
      //                \
      //                  block01 -- block02 (entry2, with height 3)
      const info = await nclient.execute('getchaintips', []);
      assert.notEqual(entry1.hash, entry2.hash);

      const expected = [
        {
          height: 3,
          hash: util.revHex(entry2.hash),
          branchlen: 3,
          status: 'valid-headers'
        },
        {
          height: 4,
          hash: util.revHex(entry1.hash),
          branchlen: 0,
          status: 'active'
        }
      ];

      // Order of chain tips is unpredictable
      try {
        assert.deepStrictEqual(info, expected);
      } catch (e) {
        assert.deepStrictEqual(info, expected.reverse());
      }
```

Reply:
Note this also might work with a single line using 

`assert.deepStrictEqual(new Set(info), new Set(expected))`